packages/core/src/httpClient/exponentialJitterBackoff.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * Calculates the delay in milliseconds for the next retry attempt. Uses "exponential backoff with
+ * full jitter" algorithm.
+ *
+ * @see https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/
+ *
+ * @param retryCount - The number of retry attempts made so far.
+ * @returns The delay in milliseconds for the next retry attempt.
+ */
+export class ExponentialJitterBackoff {
+  private readonly baseDelayMs: number;
+  private readonly maxDelayMs: number;
+  private readonly randomCallback: typeof Math.random;
+
+  /**
+   * Initializes a new instance of the ExponentialJitterBackoff class.
+   * @param baseDelayMs - The base delay in milliseconds. This will be the average delay for the first retry attempt. Defaults to 250ms.
+   * @param maxDelayMs - The maximum delay in milliseconds. This will be the maximum delay for the last retry attempt. Defaults to 15000ms.
+   * @param randomCallback - The callback to use for generating a random number between 0 and 1. Used by test only.
+   */
+  constructor(
+    baseDelayMs = 250,
+    maxDelayMs = 15000,
+    randomCallback: typeof Math.random = Math.random
+  ) {
+    this.baseDelayMs = baseDelayMs;
+    this.maxDelayMs = maxDelayMs;
+    this.randomCallback = randomCallback;
+  }
+
+  public calculateDelayInMs(retryCount: number) {
+    const exponentialDelay = this.baseDelayMs * 2 ** (retryCount + 1);
+    const cappedDelay = Math.min(exponentialDelay, this.maxDelayMs);
+
+    return Math.floor(this.randomCallback() * cappedDelay);
```

Comment:
Maybe we should consider a min cap also? Retrying immediately seems intuitively like it would not succeed, especially in the relatively common case a lot of frontends do where they will try to make a lot of request as the page loads. In this case if we assume the project gets maxed out by requests that just initiated (which is fairly easy as the max read concurrency can be as low as 3) it is unlikely that any of the other requests have finished yet if we retry immediately . Maybe a lower cap around 150ms?

Reply:
I think that might be a bad idea, because that would essentially chunk several requests together at the cap (e.g. 150 ms), potentially causing under-utilization between 0 and "cap ms" and request storm at "cap ms". This is [how its done for the Python SDK](https://github.com/cognitedata/cognite-sdk-python/blob/d38294581d13cb475e96b12f23d825dbbb1e6fb2/cognite/client/_http_client.py#L77-L83) and how the underlying article describes this so I think we want to be careful about changing this.

@alexbrasetvik your thoughts on this are welcome.

Reply:
Note I tried running the simulation with vs without a min cap and it doesn't not seem to make a statistically significant difference.